bot/src/domain/BaseEntity.ts

```
@@ -0,0 +1,95 @@
+  }
+
+  /**
+   * Default toJSON implementation that serializes all public and protected fields
+   * Can be overridden by subclasses for custom serialization
+   */
+  toJSON(): Record<string, any> {
+    const result: Record<string, any> = {};
+
+    // Get all enumerable properties
+    for (const key in this) {
+      if (this.hasOwnProperty(key)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)